pkgs/development/tools/build-managers/cmake/001-search-path.diff

```diff
@@ -22,14 +22,16 @@ index b9381c3d7d..5e944640b5 100644
    )
  
  # List common include file locations not under the common prefixes.
-+if(IS_DIRECTORY $ENV{NIX_CC}
-+  AND EXISTS $ENV{NIX_CC}/nix-support/orig-libc
-+  AND EXISTS $ENV{NIX_CC}/nix-support/orig-libc-dev)
-+  file(STRINGS "$ENV{NIX_CC}/nix-support/orig-libc" _nix_cmake_libc)
-+  file(STRINGS "$ENV{NIX_CC}/nix-support/orig-libc-dev" _nix_cmake_libc_dev)
-+else()
-+  set(_nix_cmake_libc @libc_lib@)
-+  set(_nix_cmake_libc_dev @libc_dev@)
++if(DEFINED ENV{NIX_CC})
++  if(IS_DIRECTORY "$ENV{NIX_CC}"
++    AND EXISTS "$ENV{NIX_CC}/nix-support/orig-libc"
++    AND EXISTS "$ENV{NIX_CC}/nix-support/orig-libc-dev")
++    file(STRINGS "$ENV{NIX_CC}/nix-support/orig-libc" _nix_cmake_libc)
++    file(STRINGS "$ENV{NIX_CC}/nix-support/orig-libc-dev" _nix_cmake_libc_dev)
++  else()
++    set(_nix_cmake_libc @libc_lib@)
```

Comment:
This else-clause now doesn't hit if `NIX_CC` is not set. I'm not an expert in CMake, but is it not possible to put the `DEFINED` check in the same if-statement?

Reply:
It may cause error before https://github.com/NixOS/nixpkgs/pull/192943/commits/7a9d47dea69555e7e3e7fd45895bd4f35be8e53d, but I think now it's fine!